packages/apollo/src/utils/cache.ts

```diff
@@ -0,0 +1,130 @@
+ * the cache from the previous user is not loaded to the user from persisted sources such as localStorage
+ * @example
+ * const cache = new InvalidationCache({
+ *     invalidationPolicies: {
+ *         timeToLive: 15 * 60 * 1000, // 15 minutes in milliseconds,
+ *         types: {
+ *             Contact: {
+ *                 timeToLive: 2 * 60 * 60 * 1000, // 2 hours in milliseconds,
+ *             },
+ *         },
+ *     },
+ *     cacheIdentityKey: ['ROOT_QUERY', 'authenticatedUser', 'id'],
+ * })
+ */
+export class InvalidationCache extends InvalidationPolicyCache {
+    // TODO: auth default
+    readonly cacheIdentityKey: string | Array<string> = 'id'
+
+    constructor (config?: InvalidationCacheConfig) {
+        super(config)
+
+        if (config?.cacheIdentityKey) {
+            this.cacheIdentityKey = config.cacheIdentityKey
+        }
+    }
+}
+
+/**
+ * Wrapper, which is used to create cache instance by this package logic,
+ * since specific class / implementation can be changed later
+ * @example
+ * const cache = createApolloCache({
+ *     invalidationPolicies: {
+ *         timeToLive: 15 * 60 * 1000, // 15 minutes in milliseconds,
+ *         types: {
+ *             Contact: {
+ *                 timeToLive: 2 * 60 * 60 * 1000, // 2 hours in milliseconds,
+ *             },
+ *         },
+ *     },
+ *     cacheIdentityKey: ['ROOT_QUERY', 'authenticatedUser', 'id'],
+ * })
+ */
+export function createApolloCache (config?: InvalidationCacheConfig): InvalidationCache {
+    return new InvalidationCache(config)
+}
+
+/**
+ * Merges two normalized cache object, obtained from cache.extract()
+ * @param baseCache - original cache, which will be used as baseline
+ * @param additionalCache - additional data, which will be added on top of base cache
+ */
+function mergeCaches (
+    baseCache: NormalizedCacheObject,
+    additionalCache: NormalizedCacheObject
+): NormalizedCacheObject {
+    return merge(baseCache, additionalCache, {
+        arrayMerge: (destinationArray, sourceArray) => [
+            ...sourceArray,
+            ...destinationArray.filter(destItem =>
+                sourceArray.every(srcItem => !isEqual(destItem, srcItem))
+            ),
+        ],
+    })
+}
+
+/**
+ * Extracts cache data from apollo client,
+ * when merges it together with incoming cache,
+ * and loads it back into the client
+ * @param client - apollo client to extract cache from
+ * @param cache - cache, obtained from other source (usually from persistor)
+ * @param options - merge parameters
+ * @param options.broadcast - If true, all current apollo queries will be notified and refreshed if needed
+ * @param options.overrideClient - If true, additional cache will be added on top of client's cache,
+ * otherwise - under the client's cache
+ */
+export function addCacheToClient (
+    client: ApolloClient<NormalizedCacheObject>,
+    cache: NormalizedCacheObject | undefined,
```

Comment:
Can we do it this way?
```suggestion
    cache?: NormalizedCacheObject,
```
Or this way?
```suggestion
    cache?: NormalizedCacheObject | null,
```

Reply:
We can make this one required safely :)